domehunter/dome_control.py

```diff
@@ -0,0 +1,569 @@
+        we zero the encoder at the home position, this conversion needs to
+        take into account the offset between the zero azimuth position and the
+        zero encoder position.
+
+        Parameters
+        ----------
+        az : astropy.coordinates.Longitude
+            Dome azimuth position in degrees.
+
+        Returns
+        -------
+        float
+            Returns encoder tick count corresponding to dome azimuth.
+
+        """
+        az_relative_to_home = (az - self.home_az).wrap_at(360 * u.degree)
+        return az_relative_to_home.degree / self.az_per_tick
+
+    def _ticks_to_az(self, ticks):
+        """
+        Convert encoder tick count to equivalent in degrees (azimuth).
+
+        Parameters
+        ----------
+        ticks : integer
+            The number of encoder ticks recorded.
+
+        Returns
+        -------
+        astropy.coordinates.longitude
+            The corresponding dome azimuth position in degrees.
+
+        """
+        tick_to_deg = Longitude(ticks * self.az_per_tick * u.deg)
+        return (self.home_az + tick_to_deg).wrap_at(360 * u.degree)
+
+    def _rotate_dome(self, direction):
+        """
+        Set dome to move clockwise.
+
+        Returns
+        -------
+        integer
+            Command status return code (tbd).
+
+        """
+        # if testing, deactivate the home_sernsor_pin to simulate leaving home
+        if self.testing and self.at_home:
+            self.__home_sensor_pin.drive_low()
+        # update the last_direction instance variable
+        self.last_direction = self.current_direction
+        # now update the current_direction variable to CW
+        self.current_direction = direction
+        # set the direction relay switch to CW position
+        if self.current_direction.name == "CW":
+            self.__direction_relay.on()
+            self._change_led_state(1, leds=[LED_light.RELAY_2_NO])
+            self._change_led_state(0, leds=[LED_light.RELAY_2_NC])
+        elif self.current_direction.name == "CCW":
+            self.__direction_relay.off()
+            self._change_led_state(0, leds=[LED_light.RELAY_2_NO])
+            self._change_led_state(1, leds=[LED_light.RELAY_2_NC])
+        # turn on rotation
+        self.__rotation_relay.on()
+        # update the rotation relay debug LEDs
+        self._change_led_state(1, leds=[LED_light.RELAY_1_NO])
+        self._change_led_state(0, leds=[LED_light.RELAY_1_NC])
+        cmd_status = True
+        return cmd_status
+
+    def _stop_moving(self):
+        """
+        Stop dome movement by switching the dome rotation relay off.
+        """
+        self.__rotation_relay.off()
+        # update the debug LEDs
+        self._change_led_state(0, leds=[LED_light.RELAY_1_NO])
+        self._change_led_state(1, leds=[LED_light.RELAY_1_NC])
+        # update last_direction with current_direction at time of method call
+        self.last_direction = self.current_direction
```

Comment:
Should also `self.current_direction = Direction.NONE`, I think?

Reply:
probably, although I'm kind of wondering about this... The relay can only ever be in the CW or CCW position. Does it make sense for the current direction to ever be none?